build/common/installer/scripts/tomlparser-agent-config.rb

```diff
@@ -263,17 +265,24 @@ def populateSettingValuesFromConfigMap(parsedConfig)
           puts "Using config map value: require_ack_response = #{@requireAckResponse}"
         end
       end
-      # ama-logs daemonset only settings
-      if !@controllerType.nil? && !@controllerType.empty? && @controllerType.strip.casecmp(@daemonset) == 0 && @containerType.nil?
-        # mdsd settings
-        mdsd_config = parsedConfig[:agent_settings][:mdsd_config]
-        if !mdsd_config.nil?
+
+      # mdsd settings
+      mdsd_config = parsedConfig[:agent_settings][:mdsd_config]
+      if !mdsd_config.nil?
+        # ama-logs daemonset only settings
+        if !@controllerType.nil? && !@controllerType.empty? && @controllerType.strip.casecmp(@daemonset) == 0 && @containerType.nil?
           mdsdMonitoringMaxEventRate = mdsd_config[:monitoring_max_event_rate]
           if is_valid_number?(mdsdMonitoringMaxEventRate)
             @mdsdMonitoringMaxEventRate = mdsdMonitoringMaxEventRate.to_i
             puts "Using config map value: monitoring_max_event_rate  = #{@mdsdMonitoringMaxEventRate}"
           end
         end
+
+        mdsdBackPressureThreshold = mdsd_config[:backpressure_memory_threshold]
```

Comment:
1. include the units in the config field name. For example, backpressure_memory_threshold_in_mb 
2.  Add checks to prevent customer preventing higher than limits or very small value. If in such cases, just add the message saying backpressure threshold ignored and used the default.

Reply:
1. done
2. backpressure now bounded between 100MB and container limit
3. done